tskv/src/reader/iterator.rs

```diff
@@ -396,13 +399,222 @@ pub struct FieldFileLocation {
     block_meta_iter: BlockMetaIterator,
     time_ranges: Arc<TimeRanges>,
 
-    data_block: DataBlock,
-    /// The first index of a DataType in a DataBlock
-    data_block_i: usize,
-    /// The last index of a DataType in a DataBlock
-    data_block_i_end: usize,
-    intersected_time_ranges: TimeRanges,
-    intersected_time_ranges_i: usize,
+    data_block_reader: DataBlockReader,
+}
+
+struct DataTypeWithFileId {
```

Comment:
what is the difference between the original design

Reply:
There is no difference, it just becomes easier to read.